components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+					<div class="d2l-collapsible-panel-summary">
+						<slot name="summary" @slotchange="${this._handleSummarySlotChange}"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+			</div>
+		`;
+	}
+
+	_handleActionsClick(e) {
+		const actions = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-actions');
+		if (e.target !== actions) {
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderClick(e) {
+		if (this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderSecondaryClick(e) {
+		const header = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-secondary');
+		if (e.target !== header) {
+			e.stopPropagation();
+		}
+	}
+
+	_handlePanelClick(e) {
+		if (!this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleSummarySlotChange() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('has-summary');
+	}
+
+	_onBlur() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.remove('focused');
+	}
+
+	_onFocus() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('focused');
+	}
+
+	_renderHeader() {
+		return html`
+			<div class="d2l-collapsible-panel-header" @click="${this._handleHeaderClick}">
+				<div class="d2l-collapsible-panel-header-primary">
+					${this._renderHeading()}
+					<div class="d2l-collapsible-panel-header-actions" @click="${this._handleActionsClick}">
+						<slot name="actions" @slotchange="${this._handleActionsSlotChange}"></slot>
+					</div>
+					<div class="d2l-collapsible-panel-opener">
+						<d2l-icon-custom size="tier1">
+							<svg xmlns="http://www.w3.org/2000/svg" width="10" height="18" fill="none" viewBox="0 0 10 18">
+								<path stroke="#494c4e" stroke-linejoin="round" stroke-width="2" d="m9 9-8 8V1l8 8Z"/>
+							</svg>
+						</d2l-icon-custom>
+					</div>
+				</div>
+				<div class="d2l-collapsible-panel-header-secondary" @click="${this._handleHeaderSecondaryClick}">
+					<slot name="header" @slotchange="${this._handleHeaderSlotChange}"></slot>
+				</div>
+			</div>
+			<div class="d2l-collapsible-panel-divider"></div>
+		`;
+	}
+
+	_renderHeading() {
+		const titleClasses = {
+			'd2l-collapsible-panel-title': true,
+			[`d2l-heading-${this.headingStyle}`]: true,
+		};
+
+		const headingLevel = parseInt(this.headingLevel);
+		if (headingLevel === 1) {
+			return html`<h1 class="${classMap(titleClasses)}">${this.title}</h1>`;
+		}
+		if (headingLevel === 2) {
+			return html`<h2 class="${classMap(titleClasses)}">${this.title}</h2>`;
+		}
+		if (headingLevel === 3) {
+			return html`<h3 class="${classMap(titleClasses)}">${this.title}</h3>`;
+		}
+		if (headingLevel === 4) {
+			return html`<h4 class="${classMap(titleClasses)}">${this.title}</h4>`;
+		}
+	}
+
+	_toggleExpand() {
+		this.expanded = !this.expanded;
+
+		if (!this.expanded) {
```

Comment:
There's a small bug here. Because `_toggleExpand` is only ever called when you interact with the component, if you programatically add/remove `expanded` from the host this code doesn't run, so the `scrolled` class sticks around.

A better place for all this logic is in `updated()`, which will run any time `expanded` is changed no matter how it was changed. You likely _don't_ want to fire the event though from there, so that part could stay here.